src/share/classes/com/sun/javatest/regtest/exec/RegressionScript.java

```diff
@@ -1181,12 +1182,27 @@ private String toString(List<Path> files) {
                 .map(Path::toString)
                 .collect(Collectors.joining(File.pathSeparator));
     }
-//    // where
-//    private String toString(List<File> files) {
-//        return files.stream()
-//                .map(File::getPath)
-//                .collect(Collectors.joining(File.pathSeparator));
-//    }
+
+    private Optional<String> computeMainClassNameFromTestFile() {
+        Path file = locations.absTestFile();
+        // Take file name as basis for class name.
+        String fileName = file.getFileName().toString();
+        int fileNameLength = fileName.length(); // including ".java".length() = 5
+        if (fileNameLength <= 5 && !fileName.endsWith(".java")) {
+            return Optional.empty();
+        }
+        String className = fileName.substring(0, fileNameLength - 5);
+        // Scan .java file for "package NAME;" without taking comments into account.
+        try {
+            // Assume non-large files in UTF-8 encoding.
+            String charContent = Files.readString(file);
+            Matcher packageMatcher = PACKAGE_NAME_PATTERN.matcher(charContent);
+            String packageName = packageMatcher.find() ? packageMatcher.group(1) + "." : "";
+            return Optional.of(packageName + className);
+        } catch (IOException ignored) {
```

Comment:
I think you can use "_" for the variable when its ignored, that seems to be the new convention (maybe only elsewhere) for unused variable.

Reply:
As the base line of `jtreg` is still Java 11, we cannot use a single `_` to express intentionally unused variables.